File: ts/src/seniorParkingBoy.ts
/** @format */

import {MostAvailableParkingBoyService} from './mostAvailableParkingBoyService'
import {ParkingLot} from './parkingLot'
import {ParkingBoy} from './parkingBoy'

export class SeniorParkingBoy extends ParkingBoy {
    public constructor(parkingLots: ParkingLot[]) {
        super(parkingLots, new MostAvailableParkingBoyService())
    }
}

File: kt/src/main/kotlin/com/github/scglwsj/parking/boy/SeniorParkingBoy.kt
package com.github.scglwsj.parking.boy

import com.github.scglwsj.parking.lot.ParkingLot

class SeniorParkingBoy(parkingLots: List<ParkingLot>) : ParkingBoy(parkingLots, MostAvailableParkingBoyService())
